File: 백준/Silver/9291. 스도쿠 채점/스도쿠 채점.py
import sys
input = sys.stdin.readline

test = int(input().rstrip())
for t in range(test):
    if t != 0: input().rstrip()
    maps = [list(map(int, input().rstrip().split())) for _ in range(9)]

    done = 0
    #Sero
    for i in range(9):
        S = set()
        for j in range(9):
            S.add(maps[i][j])
        
        if len(S) != 9:
            done = 1
            break

    #Garo
    if done == 0:
        for j in range(9):
            S = set()
            for i in range(9):
                S.add(maps[i][j])

            if len(S) != 9:
                done = 1
                break

        #Block
        if done == 0:
            for bi in range(3):
                for bj in range(3):
                    S = set()
                    for ai in range(3):
                        for aj in range(3):
                            S.add(maps[bi*3+ai][bj*3+aj])
                    
                    if len(S) != 9:
                        done = 1
                        break
                if done == 1: break
    
    if done == 0: print('Case %d: CORRECT'%(t+1))
    else: print('Case %d: INCORRECT'%(t+1))

File: 백준/Silver/9291. 스도쿠 채점/스도쿠 채점.cc
#include <stdio.h>

int main(void) {
    int t;
    scanf("%d", &t);

    for (int tc = 1; tc <= t; tc++) {
        int arr[9][9] = {0,};
        int done = 1;
        
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) scanf("%d", &arr[i][j]);
        }

        // Garo
        for (int i = 0; i < 9; i++) {
            int now_arr[9] = {0,};
            for (int j = 0; j < 9; j++) now_arr[arr[i][j]-1] = 1;

            for (int j = 0; j < 9; j++) {
                if (now_arr[j] == 0) {
                    done = 0;
                    break;
                }
            }
        }

        // Sero
        for (int j = 0; j < 9; j++) {
            int now_arr[9] = {0,};
            for (int i = 0; i < 9; i++) now_arr[arr[i][j]-1] = 1;

            for (int i = 0; i < 9; i++) {
                if (now_arr[i] == 0) {
                    done = 0;
                    break;
                }
            }
        }

        // Sub-Square
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                int now_arr[9] = {0,};
                for (int ni = i * 3; ni < (i + 1) * 3; ni++) {
                    for (int nj = j * 3; nj < (j + 1) * 3; nj++) now_arr[arr[ni][nj]-1] = 1;
                }

                for (int k = 0; k < 9; k++) {
                    if (now_arr[k] == 0) {
                        done = 0;
                        break;
                    }
                }
            }
        }

        if (done) printf("Case %d: CORRECT\n", tc);
        else printf("Case %d: INCORRECT\n", tc);
    }
    return 0;
}